objects/src/accounts/storage/header.rs

```diff
@@ -55,6 +58,25 @@ impl AccountStorageHeader {
             index: index as u8,
         })
     }
+
+    /// Converts storage slots of this account storage header into a vector of field elements.
+    ///
+    /// This is done by first converting each storage slot into exactly 8 elements as follows:
+    /// ```text
+    /// [STORAGE_SLOT_VALUE, storage_slot_type, 0, 0, 0]
+    /// ```
+    /// And then concatenating the resulting elements into a single vector.
+    pub fn as_elements(&self) -> Vec<Felt> {
+        self.slots
+            .iter()
+            .flat_map(|slot| {
+                let mut elements = [ZERO; 8];
+                elements[0..4].copy_from_slice(&slot.1);
+                elements[4..8].copy_from_slice(&slot.0.as_word());
+                elements
+            })
+            .collect()
```

Comment:
Instead of repeating the logic here, we should probably use `slots_as_elements()` function defined in the `super` module.

Reply:
The issue is that `slots_as_elements()` function requires the `StorageSlot` array being provided, but in case of `AccountStorageHeader` we have a vector of tuples `(StorageSlotType, Word)` instead of vector of `StrorageSlot`s.

Potentially I can change the `slots_as_elements()` function to work with tuples and implement a converter to cast `StorageSlot` into tuple, but looks like it's an overkill in that case.

Reply:
Or I can implement a wrapper type for `(StorageSlotType, Word)` tuple and implement as elements for it. Then it will be easy to cast `StorageSlot` to `StorageSlotHeader`, so we will be able to use it everywhere. This will remove the logic duplication, but we will have another type.